test/cases/resource_name/test_resource_name.py

```diff
@@ -87,11 +87,10 @@ def test_manual_in_location_at_location(self):
         """
 
         def on_chunk(chunk):
-            first, *rest = chunk
-            self.assertEqual(1, len(rest), chunk)
-            # We assume that the location span comes first, because it finishes
-            # first.
-            self.assertEqual('fuzzy.pumpkin', first['resource'], chunk)
+            self.assertEqual(2, len(chunk), chunk)
+            # We assume that the location span comes second, because it starts
```

Comment:
The original assumption made more sense to me..

Reply:
In dd-trace-cpp, a span is registered with the segment when the span is created. The segment owns the `SpanData`.